packages/editor/src/store/selectors.js

```diff
@@ -111,7 +131,15 @@ export function isEditedPostNew( state ) {
  * @return {boolean} Whether unsaved values exist.
  */
 export function isEditedPostDirty( state ) {
-	return state.editor.isDirty || inSomeHistory( state, isEditedPostDirty );
+	if ( hasChangedContent( state ) ) {
+		return true;
+	}
+
+	if ( Object.keys( state.editor.present.edits ).length > 0 ) {
+		return true;
+	}
+
+	return inSomeHistory( state, isEditedPostDirty );
```

Comment:
So if I understand properly. We conder a post dirty until it's completely saved (successful response)? right? any particular reason? should we clarify why in a comment?

Reply:
> So if I understand properly. We conder a post dirty until it's completely saved (successful response)? right? any particular reason? should we clarify why in a comment? (I know it's not new in this PR though)

It's because change detection (now only for blocks) is reset at _the start_ of the save, but we still want to avoid allowing a user to reload while that request is still pending.

It's covered in this test:

https://github.com/WordPress/gutenberg/blob/26ccc159f373e5412d4fabf20fa00431d8fce7b6/test/e2e/specs/change-detection.test.js#L216-L230

I'll see if I can form a useful comment here.

Reply:
>I'll see if I can form a useful comment here.

Added in rebased fd7a8ec.